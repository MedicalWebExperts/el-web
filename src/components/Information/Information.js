import React from 'react';
import { arrayOf, shape } from 'prop-types';
import List from '../List/List';
import { Text } from '../Typography';

const propTypes = {
  data: arrayOf(shape({})).isRequired,
};

const Information = ({ data }) => (
  <div className="information">
    <style jsx>
      {`
        .information-item {
          padding: 10px 0;
          border-bottom: 1px solid #979797;
        }

        .information-item:last-child {
          border-style: none;
        }
      `}
    </style>

    {data.map(({ title, description, list }) => (
      <div className="information-item">
        <Text align="left" color="default">
          <strong>{title}</strong>
        </Text>
        {description && (
          <Text align="left" color="default">
            {description}
          </Text>
        )}
        {list && <List data={list} />}
      </div>
    ))}
  </div>
);

Information.propTypes = propTypes;
export default Information;
