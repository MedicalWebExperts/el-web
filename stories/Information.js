import React from 'react';
import { storiesOf } from '@storybook/react';

import { Information } from '../src';

const data = [
  {
    title: 'Year Joined',
    description: '1999',
  },
  {
    title: 'Medical School',
    description: 'University of Texas Medical Branch, Galveston, TX., 1996',
  },
  {
    title: 'Residency',
    description: 'John Peter Smith Hospital, Fort Worth, TX',
  },
  {
    title: 'Board Certification',
    description: 'American Board of Family Medicine',
  },
  {
    title: 'Hospital Affilation',
    list: ['Arlington Memorial Hospital'],
  },
  {
    title: 'Professional Organizations',
    list: [
      'American Medical Association',
      'Texas Medical Association',
      'American Academy of Family Physicians',
    ],
  },
  {
    title: 'Professional Organizations',
    description: 'American Board of Family Medicine',
    list: [
      'American Medical Association',
      'Texas Medical Association',
      'American Academy of Family Physicians',
    ],
  },
];

export default () => {
  storiesOf('Information', module).add('Default', () => <Information data={data} />);
};
