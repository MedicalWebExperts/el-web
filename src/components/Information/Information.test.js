import React from 'react';
import renderer from 'react-test-renderer';
import Information from './Information';

const defaultProps = {
  data: [
    {
      title: 'Year Joined',
      description: '1999',
    },
    {
      title: 'Medical School',
      description: 'University of Texas Medical Branch, Galveston, TX., 1996',
    },
    {
      title: 'Residency',
      description: 'John Peter Smith Hospital, Fort Worth, TX',
    },
    {
      title: 'Board Certification',
      description: 'American Board of Family Medicine',
    },
    {
      title: 'Hospital Affilation',
      list: ['Arlington Memorial Hospital'],
    },
    {
      title: 'Professional Organizations',
      list: [
        'American Medical Association',
        'Texas Medical Association',
        'American Academy of Family Physicians',
      ],
    },
    {
      title: 'Professional Organizations',
      description: 'American Board of Family Medicine',
      list: [
        'American Medical Association',
        'Texas Medical Association',
        'American Academy of Family Physicians',
      ],
    },
  ],
};

describe('Component Information Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<Information {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
