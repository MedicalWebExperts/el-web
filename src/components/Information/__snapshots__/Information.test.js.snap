// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Information Snapshot without arguments, take the default values 1`] = `
<div
  className="jsx-3215497222 information"
>
  <div
    className="jsx-3215497222 information-item"
  >
    <h4
      className="jsx-610941067"
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Year Joined
    </h4>
    <h5
      className="jsx-3157562509"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "normal",
          "marginTop": 8,
        }
      }
    >
      1999
    </h5>
  </div>
  <div
    className="jsx-3215497222 information-item"
  >
    <h4
      className="jsx-610941067"
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Medical School
    </h4>
    <h5
      className="jsx-3157562509"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "normal",
          "marginTop": 8,
        }
      }
    >
      University of Texas Medical Branch, Galveston, TX., 1996
    </h5>
  </div>
  <div
    className="jsx-3215497222 information-item"
  >
    <h4
      className="jsx-610941067"
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Residency
    </h4>
    <h5
      className="jsx-3157562509"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "normal",
          "marginTop": 8,
        }
      }
    >
      John Peter Smith Hospital, Fort Worth, TX
    </h5>
  </div>
  <div
    className="jsx-3215497222 information-item"
  >
    <h4
      className="jsx-610941067"
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Board Certification
    </h4>
    <h5
      className="jsx-3157562509"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "normal",
          "marginTop": 8,
        }
      }
    >
      American Board of Family Medicine
    </h5>
  </div>
  <div
    className="jsx-3215497222 information-item"
  >
    <h4
      className="jsx-610941067"
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Hospital Affilation
    </h4>
    <ul
      className="jsx-3401328911"
      style={
        Object {
          "marginTop": 8,
        }
      }
    >
      <li
        className="jsx-1051720404"
        style={Object {}}
      >
        Arlington Memorial Hospital
      </li>
    </ul>
  </div>
  <div
    className="jsx-3215497222 information-item"
  >
    <h4
      className="jsx-610941067"
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Professional Organizations
    </h4>
    <ul
      className="jsx-3401328911"
      style={
        Object {
          "marginTop": 8,
        }
      }
    >
      <li
        className="jsx-1051720404"
        style={Object {}}
      >
        American Medical Association
      </li>
      <li
        className="jsx-1051720404"
        style={Object {}}
      >
        Texas Medical Association
      </li>
      <li
        className="jsx-1051720404"
        style={Object {}}
      >
        American Academy of Family Physicians
      </li>
    </ul>
  </div>
  <div
    className="jsx-3215497222 information-item"
  >
    <h4
      className="jsx-610941067"
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Professional Organizations
    </h4>
    <h5
      className="jsx-3157562509"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "normal",
          "marginTop": 8,
        }
      }
    >
      American Board of Family Medicine
    </h5>
    <ul
      className="jsx-3401328911"
      style={
        Object {
          "marginTop": 8,
        }
      }
    >
      <li
        className="jsx-1051720404"
        style={Object {}}
      >
        American Medical Association
      </li>
      <li
        className="jsx-1051720404"
        style={Object {}}
      >
        Texas Medical Association
      </li>
      <li
        className="jsx-1051720404"
        style={Object {}}
      >
        American Academy of Family Physicians
      </li>
    </ul>
  </div>
</div>
`;
